Shift3/boilerplate-client-react
# Refactored `Confirmation Modal`

Part of #509 

## Changes
1. A new `confirmationModalSlice` was created and added to the Rexux store.
2. All previous files under `src/common/components/ConfirmationModal` were moved to `src/features/confirmation-modal/ConfirmationModal.tsx`.
3. Added one `Global Confirmation Modal`. Previously there were separate local confirmation modals for each page.
4. The new confirmation modal was added to `AgencyListViews` `AgentListView` and `UserListView`.
5. Unit tests were added to confirmation modal.

## Purpose
In order to update the buttons with loading spinners, a refactor of the existing confirmation modal was needed. This is now  broken into 2 PRs. 

****Some of the button styling is off. Once this new confirmation modal is merged in, a second PR will address updating the buttons and styling.**

## Learning
For testing a new library was brought in `React Hooks Testing Library` for more info visit 'https://react-hooks-testing-library.com/reference/api'.

## Pre-Testing TODOs
Make sure to run the `Node Server`.

## Testing
1. Pull in the changes to your local copy of this branch and run `yarn start`.
2. Make sure the project runs and you are able to call the confirmation modal from all views.

## Screenshots
<img width="611" alt="Screen Shot 2022-01-24 at 10 55 22 AM" src="https://user-images.githubusercontent.com/59614789/150817337-76c354ce-3366-49ce-9117-f1d5ac610796.png">
